tests/app.test.js

```diff
@@ -355,6 +357,107 @@ describe('InVEST global settings: dialog interactions', () => {
+    // loaded. I've found this helps allow componentDidMount processes to
+    // finish
+    await waitFor(() => {
+      expect(nWorkersInput).toHaveValue("-1");
+      expect(loggingInput).toHaveValue("INFO");
+    });
+
+    // Change the value
+    fireEvent.change(nWorkersInput, { target: { value: nWorkers } });
+    fireEvent.change(loggingInput, { target: { value: loggingLevel } });
+    await waitFor(() => { // the value to test is inherited through props
+      expect(nWorkersInput).toHaveValue(nWorkers);
+      expect(loggingInput).toHaveValue(loggingLevel);
+    });
+    // The real test: values saved to global-settings
+    fireEvent.click(getByText('Save Changes'));
+    // Check values in the investSettingsStore
+    const investSettingsStore = testables.getSettingsStore();
+    expect(investSettingsStore.store.nWorkers).toBe('2');
+    expect(investSettingsStore.store.loggingLevel).toBe('DEBUG');
+  });
+
+  test('Load global settings', async () => {
+    const expectedSettings = {
+      nWorkers: '-1',
+      loggingLevel: 'INFO'
+    }
+    const nWorkersLabelText = 'Taskgraph n_workers parameter';
+    const loggingLabelText = 'Logging threshold';
+
+    const { getByText, getByLabelText, getByTitle } = render(
+      <App investExe="foo" />
+    );
+
+    fireEvent.click(getByTitle('settings'));
+    const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+
+    // Test that the global-settings were loaded in from store.
+    await waitFor(() => {
+      expect(nWorkersInput).toHaveValue(expectedSettings.nWorkers);
+      expect(loggingInput).toHaveValue(expectedSettings.loggingLevel);
+    });
+  });
+
+  test('Reset settings', async () => {
+    const defaultSettings = {
+      nWorkers: '-1',
+      loggingLevel: 'INFO',
+    }
+    const nWorkers = '2';
+    const loggingLevel = 'DEBUG';
+    const nWorkersLabelText = 'Taskgraph n_workers parameter';
+    const loggingLabelText = 'Logging threshold';
+
+    const { getByText, getByLabelText, getByTitle } = render(
+      <App investExe="foo" />
+    );
+
+    fireEvent.click(getByTitle('settings'));
+    const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+
+    // Check the default settings
+    await waitFor(() => {
+      // waiting because the text value depends on passed props
+      expect(nWorkersInput).toHaveValue(defaultSettings.nWorkers);
+      expect(loggingInput).toHaveValue(defaultSettings.loggingLevel);
+    });
+
+    // Change the value and reset defaults -- expect default value
+    fireEvent.change(nWorkersInput, { target: { value: nWorkers } });
+    fireEvent.change(loggingInput, { target: { value: loggingLevel } });
+    await waitFor(() => {
+      // waiting because the text value depends on passed props
```

Comment:
Same comment as above, the change handler only updates the local state, not the parent's or the store, so I think this comment is inaccurate.